src/tools.ts

```diff
@@ -27,718 +32,146 @@ class ToolException extends Error {
-            const content = input.slice(1, -1).trim();
-
-            // Split by commas
-            const pairs = content.split(',').map(pair => pair.trim());
-
-            const parsedInput: Record<string, any> = {};
-
-            for (const pair of pairs) {
-              if (!pair) continue;
-
-              // Split each pair by colon
-              const colonPos = pair.indexOf(':');
-              if (colonPos === -1) continue;
-
-              const key = pair
-                .substring(0, colonPos)
-                .trim()
-                .replace(/^["']|["']$/g, '');
-              const value = pair.substring(colonPos + 1).trim();
-
-              // Try to convert to number if possible
-              const numValue = Number(value);
-              if (!isNaN(numValue)) {
-                parsedInput[key] = numValue;
-              } else {
-                // Handle string values (remove quotes if present)
-                parsedInput[key] = value.replace(/^["']|["']$/g, '');
-              }
-            }
-
-            if (Object.keys(parsedInput).length > 0) {
-              logger.debug(
-                `Successfully parsed React agent input to: ${JSON.stringify(parsedInput)}`
-              );
-              // Call our _call method directly, bypassing schema validation
-              return this._call(parsedInput as Record<string, any>);
-            }
-          } catch (e) {
-            logger.error(`Error parsing React agent format: ${e}`);
-            // Continue to normal processing if parsing fails
-          }
-        }
-
-        // For all other inputs, use the parent class's call method for normal processing
-        // Cast to string or Record<string, any> to satisfy the type system
-        if (typeof input === 'string' || (typeof input === 'object' && input !== null)) {
-          return super.call(input as string | Record<string, any>);
-        } else {
-          // For any other input types, try to convert to string
-          return super.call(String(input));
-        }
-      } catch (error: any) {
-        // If there's an error with schema validation, try to parse it as a React agent input
-        if (error.message?.includes('validation') && typeof input === 'string') {
-          logger.debug(`Schema validation failed, trying React agent parsing as fallback`);
-          try {
-            const parsedInput = _parseReactAgentInput(input, this.schema);
-            logger.debug(`Fallback parsing succeeded: ${JSON.stringify(parsedInput)}`);
-            // Call _call directly, bypassing validation
-            return this._call(parsedInput);
-          } catch (fallbackError) {
-            logger.error(`Fallback parsing also failed: ${fallbackError}`);
-          }
-        }
-
-        // Re-throw the original error if all attempts fail
-        throw error;
-      }
+  tool: { name: string; description?: string; inputSchema?: any },
+  toolSchema?: any,
+  responseFormat: ResponseFormat = 'text'
```

Comment:
should the default actually be `content_and_artifact`? the only thing it will change is the `toolMessage.artifact` field after the tool is invoked - content will be unchanged

Reply:
done